wakuv2/filter_manager.go

```diff
@@ -0,0 +1,360 @@
+		mgr.eventChan <- FilterEvent{eventType: FilterEventSubscribeResult, filterID: filterID, tempID: tempID, result: false}
+		return
+	}
+	contentFilter := mgr.buildContentFilter(f.PubsubTopic, f.ContentTopics)
+	mgr.logger.Debug("FILTER subscribe to filter node", zap.Stringer("peer", peer), zap.String("pubsubTopic", contentFilter.PubsubTopic), zap.Strings("contentTopics", contentFilter.ContentTopicsList()))
+	ctx, cancel := context.WithTimeout(mgr.ctx, requestTimeout)
+	defer cancel()
+
+	subDetails, err := mgr.node.FilterLightnode().Subscribe(ctx, contentFilter, filter.WithPeer(peer))
+	success := err == nil
+	var sub *subscription.SubscriptionDetails
+	if err != nil {
+		mgr.logger.Warn("FILTER could not add wakuv2 filter for peer", zap.String("filterId", filterID), zap.Stringer("peer", peer), zap.Error(err))
+	} else {
+		mgr.logger.Debug("FILTER subscription success", zap.String("filterId", filterID), zap.Stringer("peer", peer), zap.String("pubsubTopic", contentFilter.PubsubTopic), zap.Strings("contentTopics", contentFilter.ContentTopicsList()))
+		sub = subDetails[0]
+
+	}
+
+	mgr.eventChan <- FilterEvent{eventType: FilterEventSubscribeResult, filterID: filterID, tempID: tempID, sub: sub, result: success}
+}
+
+func (mgr *FilterManager) unsubscribeFromFilter(filterID string, sub *subscription.SubscriptionDetails) {
+	mgr.logger.Debug("FILTER unsubscribe from filter node", zap.String("filterId", filterID), zap.String("subId", sub.ID), zap.Stringer("peer", sub.PeerID))
+	// Unsubscribe on light node
+	ctx, cancel := context.WithTimeout(mgr.ctx, requestTimeout)
+	defer cancel()
+	_, err := mgr.node.FilterLightnode().UnsubscribeWithSubscription(ctx, sub)
+	success := err == nil
+
+	if err != nil {
+		mgr.logger.Warn("could not unsubscribe wakuv2 filter for peer", zap.String("filterId", filterID), zap.String("subId", sub.ID), zap.Error(err))
+	}
+
+	mgr.eventChan <- FilterEvent{eventType: FilterEventUnsubscribeResult, filterID: filterID, result: success, sub: sub}
+}
+
+// Check whether each of the installed filters
+// has enough alive subscriptions to peers
+func (mgr *FilterManager) pingPeers() {
+	mgr.logger.Debug("FILTER pingPeers")
+
+	distinctPeers := make(map[peer.ID]struct{})
+	for filterID, subs := range mgr.filterSubs {
+		if len(subs) == 0 { // TODO !!!!!!!
+			// No subs found, trigger full resubscribe
+			mgr.logger.Debug("FILTER ping peer no subs", zap.String("filterId", filterID))
+			go func() {
+				mgr.eventChan <- FilterEvent{eventType: FilterEventPingResult, filterID: filterID, result: false}
+			}()
+			continue
+		}
+		for _, sub := range subs {
+			if sub == nil {
+				// Skip temp subs
+				continue
+			}
+			_, found := distinctPeers[sub.PeerID]
+			if found {
+				continue
+			}
+			distinctPeers[sub.PeerID] = struct{}{}
+			mgr.logger.Debug("FILTER ping peer", zap.Stringer("peerId", sub.PeerID))
+			go func(sub *subscription.SubscriptionDetails) {
+				err := mgr.isFilterSubAlive(sub)
```

Comment:
matching the exact error would feel safer here :) but maybe there's many and they all indicate a dead connection?

Reply:
Yes, we don't really care about the error type, just if it's there or not. Maybe @richard-ramos can weight in and correct my understanding, if necessary.